python/dgl/sparse/sparse_matrix.py

```diff
@@ -586,6 +586,75 @@ def range_select(self, dim: int, index: slice):
             )
         raise TypeError(f"{type(index).__name__} is unsupported input type.")
 
+    def sample(
+        self, dim: int, n_pick: int, replacement: Optional[bool] = False
```

Comment:
I think the current DGL implementation supports sampling on some rows/columns and sampling bias. We may need to change this interface a bit.
```suggestion
        self, dim: int, n_pick: int, ids: Optional[torch.Tensor] = None, replacement: Optional[bool] = False, bias: Optional[bool] = False
```

`ids` indicates the row/columns ids to sample and `ids = None` means sampling on the whole matrix. Then this function may change the shape of a sparse matrix. `bias` means whether use the value of the sparse matrix as sampling bias or not.

Reply:
OK, I find this too when implement the API